FlowCrypt/src/test/java/com/flowcrypt/email/core/msg/MsgBlockParserTest.kt

```diff
@@ -0,0 +1,280 @@
+/*
+ * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
+ * Contributors: Ivan Pizhenko
+ */
+
+package com.flowcrypt.email.core.msg
+
+import com.flowcrypt.email.api.retrofit.response.model.node.GenericMsgBlock
+import com.flowcrypt.email.api.retrofit.response.model.node.MsgBlock
+import com.flowcrypt.email.api.retrofit.response.model.node.PublicKeyMsgBlock
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class MsgBlockParserTest {
+  @Test
+  fun testDetectBlocksDoesNotGetTrippedOnBlocksWithUnknownHeaders() {
+    checkForSinglePlaintextBlock(
+        "This text breaks email and Gmail web app.\n\n" +
+        "-----BEGIN FOO-----\n\n" +
+        "Even though it's not a valid PGP m\n\n" +
+        "Muhahah"
+    )
+  }
+
+  @Test
+  fun testDetectBlocksIgnoresFalsePositiveBlocks() {
+    checkForSinglePlaintextBlock(
```

Comment:
Maybe it'll be better to store such a long string as resources. Such code looks bloated and a little hard to read it.

Reply:
I think it's ok for unit tests, we do it similarly same in the other repos too.